client/lib/plans/test/can-upgrade-to-plan.js

```diff
@@ -0,0 +1,116 @@
+/** @format */
+/**
+ * External Dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal Dependencies
+ */
+import { canUpgradeToPlan } from '..';
+import {
+	PLAN_BUSINESS,
+	PLAN_FREE,
+	PLAN_JETPACK_BUSINESS,
+	PLAN_JETPACK_BUSINESS_MONTHLY,
+	PLAN_JETPACK_FREE,
+	PLAN_JETPACK_PERSONAL,
+	PLAN_JETPACK_PERSONAL_MONTHLY,
+	PLAN_JETPACK_PREMIUM,
+	PLAN_JETPACK_PREMIUM_MONTHLY,
+	PLAN_PERSONAL,
+	PLAN_PREMIUM,
+} from '../constants';
+
+describe( 'canUpgradeToPlan', () => {
+	it( 'should return true from lower-tier plans to higher-tier plans', () => {
+		const makeSite = product_slug => ( {
```

Comment:
Also, while I understand naming the argument `product_slug` makes the code shorter, it's not compliant to our coding standards, so we should probably consider renaming it to `productSlug`.